advection/conservative_sw_mhd.jl

```diff
@@ -26,13 +27,27 @@ model = ShallowWaterModel(grid = grid,
                           formulation = ConservativeFormulation()
                           )
 
-Aᵢ(x, y, z) = -0.05*y
+Aᵢ(x, y, z) = 0.1*exp(-((x - 0.5)^2 + y^2)) - 0.1*exp(-((x + 0.5)^2 + y^2))
 
 hᵢ(x, y, z) = 1
-uhᵢ(x, y, z) = y*exp(-(x^2 + y^2))
-vhᵢ(x, y, z) = -x*exp(-(x^2 + y^2))
-set!(model, uh = uhᵢ, vh = vhᵢ, h = hᵢ, A = Aᵢ)
-simulation = Simulation(model, Δt = 0.01, stop_time = 25)
+#uhᵢ(x, y, z) = y*exp(-(x^2 + y^2))
+#vhᵢ(x, y, z) = -x*exp(-(x^2 + y^2))
+set!(model,#= uh = uhᵢ, vh = vhᵢ,=# h = hᵢ, A = Aᵢ)
+simulation = Simulation(model, Δt = 0.01, stop_time = 70)
+
+uh, vh, h = simulation.model.solution
+u = uh / h
+v = vh / h
+A = simulation.model.tracers.A
+B_x = -∂y(A) / h
+B_y = ∂x(A) / h
+kinetic_term = h*(u^2 + v^2)
+magnetic_term = h*(B_x^2 + B_y^2)
+pressure_term = 9.81*h^2
```

Comment:
shallow water has a parameter `g` in the model.  Let's use that instead of hardcoding 9.81 in to this term.  I think you can use `model.graviational_accelration.g`, or something like that.